charts/pravega/templates/pravega.yaml

```diff
@@ -5,43 +5,106 @@ metadata:
+        memory: {{ .Values.controller.resources.requests.memory | quote }}
+        cpu: {{ .Values.controller.resources.requests.cpu | quote }}
       limits:
-        memory: "3Gi"
-        cpu: "2000m"
-    segmentStoreReplicas: {{ .Values.pravega.segmentStoreReplicas }}
+        memory: {{ .Values.controller.resources.limits.memory | quote }}
+        cpu: {{ .Values.controller.resources.limits.cpu | quote }}
+    segmentStoreReplicas: {{ .Values.segmentStore.replicas }}
     segmentStoreResources:
       requests:
-        memory: "3Gi"
-        cpu: "1000m"
+        memory: {{ .Values.segmentStore.resources.requests.memory | quote }}
+        cpu: {{ .Values.segmentStore.resources.requests.cpu | quote }}
       limits:
-        memory: "5Gi"
-        cpu: "2000m"
-    debugLogging: {{ .Values.pravega.debugLogging }}
+        memory: {{ .Values.segmentStore.resources.limits.memory | quote }}
+        cpu: {{ .Values.segmentStore.resources.limits.cpu | quote }}
+    {{- if .Values.segmentStore.env }}
+    segmentStoreEnvVars: {{ .Values.segmentStore.env }}
+    {{- end }}
+    {{- if .Values.segmentStore.secret }}
+    segmentStoreSecret:
+      {{- if .Values.segmentStore.secret.name }}
+      secret: {{ .Values.segmentStore.name }}
+      {{- end }}
+      {{- if .Values.segmentStore.secret.path }}
+      mountPath: {{ .Values.segmentStore.path }}
+      {{- end }}
+    {{- end }}
+    controllerjvmOptions: {{ .Values.controller.jvmOptions }}
+    segmentStoreJVMOptions: {{ .Values.segmentStore.jvmOptions }}
+    debugLogging: {{ .Values.debugLogging }}
     cacheVolumeClaimTemplate:
       accessModes: [ "ReadWriteOnce" ]
-      storageClassName: {{ .Values.pravega.storage.className }}
+      storageClassName: {{ .Values.storage.cache.className }}
       resources:
         requests:
-          storage: {{ .Values.pravega.cacheVolumeRequest }}
+          storage: {{ .Values.storage.cache.size }}
     longtermStorage:
+      {{- if eq .Values.storage.longtermStorage.type "ecs"}}
+      ecs:
+        configUri: {{ .Values.storage.longtermStorage.ecs.configUri }}
+        bucket: {{ .Values.storage.longtermStorage.ecs.bucket }}
+        prefix: {{ .Values.storage.longtermStorage.ecs.prefix }}
+        credentials: {{ .Values.storage.longtermStorage.ecs.credentials }}
+      {{- else if eq .Values.storage.longtermStorage.type "hdfs"}}
+      hdfs:
+        uri: {{ .Values.storage.longtermStorage.hdfs.uri }}
+        root: {{ .Values.storage.longtermStorage.hdfs.root }}
+        replicationFactor: {{ .Values.storage.longtermStorage.hdfs.replicationFactor }}
+      {{- else }}
       filesystem:
         persistentVolumeClaim:
-          claimName: {{ .Values.pravega.longtermStorage }}
+          claimName: {{ .Values.storage.longtermStorage.filesystem.pvc }}
+      {{- end }}
+    {{- if .Values.options.enabled }}
+    options:
+      {{- range .Values.options.list }}
+      {{ .key }} : {{ .value | quote }}
+      {{- end }}
```

Comment:
Might be easier to do:
```
    options:
      {{- if .Values.pravega_options }}
{{ toYaml .Values.pravega_options | indent 6 }}
      {{- end }}
```

That way the values would be a bit simpler to write, e.g.:
```
pravega_options:
  controller.auth.userPasswordFile: "/etc/auth-passwd-volume/password-file.txt"
  controller.auth.tlsEnabled: "false"
  controller.auth.tlsCertFile: "/etc/secret-volume/tls.crt"
  controller.auth.tlsTrustStore: "/etc/secret-volume/tls.crt"
```

Reply:
By specifying options in this way, we get to set the options in the following way :-
```
helm install <name> charts/pravega --set options.enabled=true --set options.list[0].key=bookkeeper.bkLedgerPath --set options.list[0].value="/pravega/bar-pravega/bookkeeper/ledgers"
```
which was a requirement as per #393